client.go

```diff
@@ -171,14 +176,22 @@ func (c *Client) run() {
 			call.errs <- c.recv(call.resp, msg)
 			delete(waiters, msg.StreamID)
 		case <-shutdown:
-			shutdownErr = errors.Wrapf(shutdownErr, "ttrpc: client shutting down")
+			if shutdownErr != nil && shutdownErr != io.EOF {
```

Comment:
You could check for `EPIPE` here as well

Reply:
EOF comes from `read()` and EPIPE would come from a `write()` on a closed connection

Reply:
I am experimenting with getting that right. Unfortunately, it takes a bit of effort to reproduce.